python/ray/_private/accelerator.py

```diff
@@ -0,0 +1,94 @@
+import json
+import os
+import subprocess
+import sys
+
+import ray._private.utils
+import ray._private.ray_constants as ray_constants
+
+
+def update_resources_with_accelerator_type(resources, num_gpus):
+    """Update the resources dictionary with the accelerator type and custom
+    resources.
+    """
+    _detect_and_configure_aws_neuron_core(resources, num_gpus)
+
+
+def _detect_and_configure_aws_neuron_core(resources, num_gpus):
+    """Configuration and auto-detection of AWS NeuronCore accelerator type
+    and number of NeuronCore.
+
+    If the number of NeuronCore is not specified in the resources, this
+    function will try to detect the number of NeuronCore.
+
+    If the number of NeuronCore is specified in the resources, this
+    function will check if the number of NeuronCore is greater than the
+    number of visible NeuronCore.
+
+    Also, throws an error if GPUs are configured along with NeuronCores.
+    """
+    # AWS NeuronCore detection and configuration
+    # 1. Check if the user specified num_neuron_cores in resources
+    num_neuron_cores = resources.get(ray_constants.NUM_NEURON_CORES, None)
+    # 2. Check if the user specified NEURON_RT_VISIBLE_CORES
+    neuron_core_ids = ray._private.utils.get_aws_neuron_core_visible_ids()
+    if (
+        num_neuron_cores is not None
+        and neuron_core_ids is not None
+        and num_neuron_cores > len(neuron_core_ids)
+    ):
+        raise ValueError(
+            f"Attempting to start raylet with {num_neuron_cores} "
+            f"neuron cores, but NEURON_RT_VISIBLE_CORES contains "
+            f"{neuron_core_ids}."
+        )
+    # 3. Auto-detect num_neuron_cores if not specified in resources
+    if num_neuron_cores is None:
+        num_neuron_cores = _autodetect_aws_neuron_cores()
+        # Don't use more neuron cores than allowed by NEURON_RT_VISIBLE_CORES.
+        if None not in (neuron_core_ids, num_neuron_cores):
+            num_neuron_cores = min(num_neuron_cores, len(neuron_core_ids))
+    if num_neuron_cores is not None:
+        if num_gpus is not None and num_gpus > 0:
```

Comment:
Great Q. Fixed it by adding the validation on remote options. Yes, a Ray cluster can have heterogenous resources (GPU + NeuronCore + ...) but a remote function/work can only run on either on GPU or NeuronCore because the underlying resource can only contain one or the other.